src/digraph.rs

```diff
@@ -295,17 +295,20 @@ impl PyDiGraph {
 #[pymethods]
 impl PyDiGraph {
     #[new]
-    #[pyo3(signature=(/, check_cycle=false, multigraph=true, attrs=None, *, initial_node_count=0, initial_edge_count=0))]
+    #[pyo3(signature=(/, check_cycle=false, multigraph=true, attrs=None, *, node_count_hint=None, edge_count_hint=None))]
     fn new(
         py: Python,
         check_cycle: bool,
         multigraph: bool,
         attrs: Option<PyObject>,
-        initial_node_count: usize,
-        initial_edge_count: usize,
+        node_count_hint: Option<usize>,
+        edge_count_hint: Option<usize>,
```

Comment:
TBH, I'm not the biggest fan of this change. There doesn't seem to be a difference between `Some(0)` and `None`  and I feel it makes it simpler both for python and rust to just default to `0`. I'd personally prefer if we just left the type as a `usize` instead of an `Option<usize>`. That being said it's super minor and won't effect the runtime performance or anything so I don't feel strongly enough to block on this.

Reply:
I agree but my main argument for `None` is that maybe one day we could want to change. Something more along the lines of small vector optimizations. With `None` I know we can override it to whatever default it sets. With `0`, I think it would be poor taste to not allocate a vector with zero capacity.